src/Pages/GlobalConfigurations/Authorization/shared/components/AppPermissions/types.ts

```diff
@@ -0,0 +1,43 @@
+import { ServerError } from '@devtron-labs/devtron-fe-common-lib'
+import { JobList } from '../../../../../../components/Jobs/Types'
+import { ACCESS_TYPE_MAP } from '../../../../../../config'
+import { DirectPermissionsRoleFilter } from '../../../types'
+
+type AppsList = Map<number, { loading: boolean; result: { id: number; name: string }[]; error: ServerError }>
+type JobsList = Map<number, { loading: boolean; result: JobList['result']['jobContainers']; error: ServerError }>
+
+export interface AppPermissionsDetailType {
+    accessType: ACCESS_TYPE_MAP.DEVTRON_APPS | ACCESS_TYPE_MAP.HELM_APPS | ACCESS_TYPE_MAP.JOBS
+    handleDirectPermissionChange: (...rest) => void
+    removeDirectPermissionRow: (index: number) => void
+    addNewPermissionRow: (
+        accessType: ACCESS_TYPE_MAP.DEVTRON_APPS | ACCESS_TYPE_MAP.HELM_APPS | ACCESS_TYPE_MAP.JOBS,
+    ) => void
+    appsListHelmApps: AppsList
+    jobsList: JobsList
+    appsList: AppsList
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    getEnvironmentOptions: any
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    projectsList: any[]
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    environmentClusterOptions: any
+    getListForAccessType: (accessType: ACCESS_TYPE_MAP) => AppsList | JobsList
+}
+
+export interface DirectPermissionRow
+    extends Pick<
```

Comment:
Can we opt for omit instead of pick?

Reply:
I wanted to be specific on what props should be added to the type. With omit we might not have clarity on what props are being picked